File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}

        for idx, val in enumerate(nums):
            diff = target - val
            if diff in dic:
                return [dic[diff], idx]
            dic[val] = idx 

        return []

File: Data Structures & Algorithms/remove-element/submission-0.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int k = 0;
        int n = nums.length;
        while (k < n) {
            if(nums[k] == val) {
                n--;
                nums[k] = nums[n];
            } else {
                k++;
            }
        }
        return n;
    }
}
